pkg/controllers/provisioning/v1alpha1/reallocator/controller.go

```diff
@@ -13,3 +13,67 @@ limitations under the License.
 */
 
 package reallocator
+
+import (
+	"context"
+	"fmt"
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	"github.com/awslabs/karpenter/pkg/controllers"
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+// Controller for the resource
+type Controller struct {
+	filter        *Filter
+	cloudProvider cloudprovider.Factory
+}
+
+// For returns the resource this controller is for.
+func (c *Controller) For() controllers.Object {
+	return &v1alpha1.Provisioner{}
+}
+
+// Owns returns the resources owned by this controller's resource.
+func (c *Controller) Owns() []controllers.Object {
+	return []controllers.Object{}
+}
+
+func (c *Controller) Interval() time.Duration {
+	return 5 * time.Second
+}
+
+// NewController constructs a controller instance
+func NewController(kubeClient client.Client, cloudProvider cloudprovider.Factory) *Controller {
+	return &Controller{
+		filter:        &Filter{kubeClient: kubeClient},
+		cloudProvider: cloudProvider,
+	}
+}
+
+// Reconcile executes an allocation control loop for the resource
+func (c *Controller) Reconcile(object controllers.Object) error {
+	ctx := context.TODO()
```

Comment:
Are you up for wiring https://github.com/awslabs/karpenter/blob/main/pkg/controllers/controller.go#L67 through? It's been copied around enough times. We should use the ctx given to us by controller runtime. Doesn't have to be this PR.

Reply:
I'll look into it while I'm doing my tasks 👍